test/functional/pdo_sqlsrv/PDO38_FetchBound.phpt

```diff
@@ -40,72 +34,78 @@ function Fetch()
-    InsertRowEx($conn1, $tableName, $dataCols, "2, 'String2'", null);
+    createTable($conn1, $tableName, array(new ColumnMeta("int", "idx", "NOT NULL PRIMARY KEY"), "txt" => "varchar(20)"));
+    insertRow($conn1, $tableName, array("idx" => 0, "txt" => 'String0'));
+    insertRow($conn1, $tableName, array("idx" => 1, "txt" => 'String1'));
+    insertRow($conn1, $tableName, array("idx" => 2, "txt" => 'String2'));
 
     // Testing with prepared query
-    $stmt1 = PrepareQuery($conn1, "SELECT COUNT(idx) FROM [$tableName]");
-
+    $stmt1 = $conn1->prepare("SELECT COUNT(idx) FROM [$tableName]");
     $stmt1->execute();
     var_dump($stmt1->fetchColumn());
     unset($stmt1);
 
-    $stmt1 = PrepareQuery($conn1, "SELECT idx, txt FROM [$tableName] ORDER BY idx");
+    if (!isColEncrypted()) {
+        $stmt1 = PrepareQuery($conn1, "SELECT idx, txt FROM [$tableName] ORDER BY idx");
+    } else {
+        // ORDER BY does not work on encrypted columns
+        $stmt1 = PrepareQuery($conn1, "SELECT idx, txt FROM [$tableName]");
+    }
     $stmt1->execute();
     $data = $stmt1->fetchAll(PDO::FETCH_COLUMN|PDO::FETCH_UNIQUE);
+    if (isColEncrypted()) {
+        ksort($data);
+    }
     var_dump($data);
 
     echo "===WHILE===\n";
 
     $stmt1->bindColumn('idx', $idx);
     $stmt1->bindColumn('txt', $txt);
     $stmt1->execute();
-    while ($stmt1->fetch(PDO::FETCH_BOUND))
-    {
-        var_dump(array($idx=>$txt));
+    $boundCols = array();
+    while ($stmt1->fetch(PDO::FETCH_BOUND)) {
+        $boundCols[$idx] = $txt;
+    }
+    if (isColEncrypted()) {
+        ksort($boundCols);
     }
+    var_dump($boundCols);
 
     echo "===ALONE===\n";
 
-    $stmt2 = PrepareQuery($conn1, "SELECT txt FROM [$tableName] WHERE idx=:inp");
+    $stmt2 = $conn1->prepare("SELECT txt FROM [$tableName] WHERE idx=:inp");
     $stmt2->bindParam(':inp', $idx);    // by foreign name
 
-    $stmt3 = PrepareQuery($conn1, "SELECT idx FROM [$tableName] WHERE txt=:txt");
+    $stmt3 = $conn1->prepare("SELECT idx FROM [$tableName] WHERE txt=:txt");
     $stmt3->bindParam(':txt', $txt);    // using same name
 
-    foreach($data as $idx => $txt)
-    {
+    foreach ($data as $idx => $txt) {
         var_dump(array($idx=>$txt));
 
         var_dump($stmt2->execute());
-        if ($idx == 0)
-        {   // bindColumn()s after execute() has been called at least once
+        if ($idx == 0) {   // bindColumn()s after execute() has been called at least once
             $stmt2->bindColumn('txt', $col1);
         }
         var_dump($stmt2->fetch(PDO::FETCH_BOUND));
         $stmt2->closeCursor();
 
         var_dump($stmt3->execute());
-        if ($idx == 0)
-        {   // bindColumn()s after execute() has been called at least once
+        if ($idx == 0) {   // bindColumn()s after execute() has been called at least once
             $stmt3->bindColumn('idx', $col2);
         }
         var_dump($stmt3->fetch(PDO::FETCH_BOUND));
```

Comment:
please use more descriptive variables than $col1 or $col2 (because one is for txt column and the other for idx, which is actually the first column in the table)